numpy/lib/type_check.py

```diff
@@ -444,10 +465,17 @@ def nan_to_num(x, copy=True):
 
     dest = (x.real, x.imag) if iscomplex else (x,)
     maxf, minf = _getmaxmin(x.real.dtype)
+    if posinf is not None:
+        maxf = posinf
+    if neginf is not None:
+        minf = neginf
     for d in dest:
-        _nx.copyto(d, 0.0, where=isnan(d))
-        _nx.copyto(d, maxf, where=isposinf(d))
-        _nx.copyto(d, minf, where=isneginf(d))
+        idx_nan = isnan(d)
+        idx_posinf = isposinf(d)
+        idx_neginf = isneginf(d)
+        _nx.copyto(d, nan, where=idx_nan)
+        _nx.copyto(d, maxf, where=idx_posinf)
+        _nx.copyto(d, minf, where=idx_neginf)
```

Comment:
I don't see the point of splitting these 3 lines into 6?

Reply:
This is to maintain what the user has chosen. If there is only this:

        ```
        _nx.copyto(d, nan, where=idx_nan)
        _nx.copyto(d, maxf, where=idx_posinf)
        _nx.copyto(d, minf, where=idx_neginf)
```
If in the keywords one chooses `nan=np.inf` and `posinf=-999` the following array:
`a = [np.nan, 10, np.inf]`
would be converted to:
`a = [-999, 10, -999]`
So the `nan` keyword would be overwritten as it is the first being executed. I think this wouldn't be the expected behavior. I would expect:
`a = [np.inf, 10, -999]`

This way it is necessary to create temporary boolean arrays but I think it makes more sense.

Reply:
Also, this is why I've added these tests:

https://github.com/numpy/numpy/pull/13219/files/f486ca24f20ca5d4de43191ef2d647c87c37ac1a#diff-c04fcf6a62bc1bda31fd3a1486786567R450

Reply:
Oh right, that makes sense, thanks